app/services/earnings_csv_parser.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+require_relative '../models/earning'  # Add this line if Earning model is not being loaded
+
+class EarningsCsvParser
+  STRING = 'STRING'.freeze
+  DECIMAL = 'DECIMAL'.freeze
+  # file = CSV string
+  def initialize(file, employer_id)
+      @file = file
+      @employer = get_employer(employer_id)
+      @employer_csv_format = get_employer_csv_format
+  end
+
+  def create_earnings
+      csv = CSV.parse(@file, headers: true)
+      errors = []
+      csv.each do |row|
+        e = create_earning(row)
+        if e.valid?
+          e.save
+        else
+          errors << "Error with row #{row}: #{e.errors.full_messages}"
+        end
```

Comment:
`e.save` will return a boolean if the record is valid and was saved, so you can simplify to something like this:

```suggestion
        unless e.save
          errors << "Error with row #{row}: #{e.errors.full_messages}"
        end
```

Reply:
That's much cleaner - good suggestion